astropy/stats/jackknife.py

```diff
@@ -0,0 +1,197 @@
+
+    Jackknife resampling is a technique to generate 'n' deterministic samples
+    of size 'n-1' from a measured sample of size 'n'. Basically, the i-th
+    sample, (1<=i<=n), is generated by means of removing the i-th measurement
+    of the original sample. Like the bootstrap resampling, this statistical
+    technique finds applications in estimating variance, bias, and confidence
+    intervals.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Original sample (1-D array) from which the jackknife resamples will be
+        generated.
+
+    Returns
+    -------
+    resamples : numpy.ndarray
+        The i-th row is the i-th jackknife sample, i.e., the original sample
+        with the i-th measurement deleted.
+
+    Examples
+    --------
+    1. Obtain Jackknife resamples:
+
+    >>> import numpy as np
+    >>> from astropy.stats import jackknife_resampling
+    >>> data = np.array([1,2,3,4,5,6,7,8,9,0])
+    >>> resamples = jackknife_resampling(data)
+    >>> resamples
+    array([[ 2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  3.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  4.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  5.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  6.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  7.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  8.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  9.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  0.],
+           [ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.]])
+    >>> resamples.shape
+    (10, 9)
+
+    References
+    ----------
+    .. [1] McIntosh, Avery. "The Jackknife Estimation Method".
+        <http://people.bu.edu/aimcinto/jackknife.pdf>
+
+    .. [2] Efron, Bradley. "The Jackknife, the Bootstrap, and other
+        Resampling Plans". Technical Report No. 63, Division of Biostatistics,
+        Stanford University, December, 1980.
+
+    .. [3] Cowles, Kate. "Computing in Statistics: The Jackknife, Lecture 11".
+        <http://homepage.stat.uiowa.edu/~kcowles/s166_2009/lect11.pdf>.
+        September, 2009.
+    """
+
+    n = data.shape[0]
+    assert n > 0, "data must contain at least one measurement"
+
+    resamples = np.empty([n, n-1])
+
+    for i in range(n):
+        resamples[i] = np.delete(data, i)
+
+    return resamples
+
+
+def jackknife_stats(data, statistic, conf_lvl=0.95):
+    """ Performs jackknife estimation on the basis of jackknife resamples.
+
+    This function requires `SciPy <http://www.scipy.org>`_ to be installed.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Original sample (1-D array).
+    statistic : function
+        Any function (or vector of functions) on the basis of the measured
+        data, e.g, sample mean, sample variance, etc. The jackknife estimate of
+        this statistic will be returned.
+    conf_lvl : float
```

Comment:
Since this is an optional parameter you could include that it is ``optional`` here and mention the default value in the parameter description.